edc-extensions/agreements/retirement-evaluation-core/src/main/java/org/eclipse/tractusx/edc/agreements/retirement/defaults/DefaultAgreementsRetirementService.java

```diff
@@ -0,0 +1,87 @@
+/********************************************************************************
+ * Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+package org.eclipse.tractusx.edc.agreements.retirement.defaults;
+
+import org.eclipse.edc.connector.controlplane.contract.spi.types.agreement.ContractAgreement;
+import org.eclipse.edc.policy.engine.spi.PolicyContext;
+import org.eclipse.edc.spi.query.Criterion;
+import org.eclipse.edc.spi.query.QuerySpec;
+import org.eclipse.edc.spi.result.ServiceResult;
+import org.eclipse.edc.transaction.spi.TransactionContext;
+import org.eclipse.tractusx.edc.agreements.retirement.spi.service.AgreementsRetirementService;
+import org.eclipse.tractusx.edc.agreements.retirement.spi.store.AgreementsRetirementStore;
+import org.eclipse.tractusx.edc.agreements.retirement.spi.types.AgreementsRetirementEntry;
+
+import java.util.List;
+
+/**
+ * Default implementation for a {@link AgreementsRetirementService}.
+ */
+public class DefaultAgreementsRetirementService implements AgreementsRetirementService {
+
+    private final AgreementsRetirementStore store;
+    private final TransactionContext transactionContext;
+
+    public DefaultAgreementsRetirementService(AgreementsRetirementStore store, TransactionContext transactionContext) {
+        this.store = store;
+        this.transactionContext = transactionContext;
+    }
+
+    @Override
+    public boolean isRetired(PolicyContext context) {
```

Comment:
as stated in the interface i would use `ContractAgreement` and let the caller handle the failure within the policy evaluation